### PR TITLE
Plugins Autoconfig: Update notice text for users with already-registered plugins

### PR DESCRIPTION
**Pull request by ryelle**
This PR continues discussion from #9703. I've merged the code to handle duplicate plugin registration, this PR is to discuss what the copy and action should be.

I've also take this opportunity to take @gwwar's advice and refactor `renderStatus` into `renderStatus` and `renderStatusError`, so the logic should be easier to follow.

To test (same as previous PR)

- Set up Akismet under one wp.com account
- Connect Jetpack on that site to a different wp.com account, and buy a plan
- Go through the autoconfig flow (should be prompted after buying plan, or you can go directly to /plugins/setup/$site)
- The plugins will run through installing, and once everything's finished, you should see a blue notice under Akismet, saying "This plugin is already registered with another plan.", with an action to "Manage Plans" linking to `/me/purchases`

screen shot 2016-12-08 at 10 33 14 am

(I've marked this in progress as I've noticed an error with the plugin notices, but I want to keep the convo going on what this copy should be.)

cc @johnHackworth @rickybanister @beaulebens @richardmuscat @annezazuu

### CODE REVIEW

**Comment by ryelle** [comment]
In the last convo, @rickybanister said:

> I'm somewhat confused by the 'this plugin is already configured with another plan' language. We don't want to overwrite their current configuration, but we also don't want them paying twice.
>
> I think we can avoid this situation better further upstream by making the plans page more aware of existing old-style plans, but if they do get to this point perhaps we should be slightly more proactive.
> 
> Can we have that notice say something more like `Akismet was already configured | Manage Plans?`

"This plugin was already configured" doesn't seem like a problem I need to fix. I kept "This plugin is already registered with another plan" because it clearly (to me) states the problem.

I've added the Manage Plans action, which here is linking to `/me/purchases`, but I'm not sure if that's where you intended that to go? In the case of the older dops plans, I think the user will need to bounce between product dashboards, or chat with an HE.

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx [hunk 1]
```diff
@@ -72,6 +72,10 @@ const PlansSetup = React.createClass( {
 		analytics.tracks.recordEvent( 'calypso_plans_autoconfig_click_manual_error' );
 	},
 
+	trackManagePlans() {
+		analytics.tracks.recordEvent( 'calypso_plans_autoconfig_click_manage_plans' );
+	},
```
Comment (gwwar) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: Let's leave this for another PR, but we prefer to use the analytics redux action when possible. This makes it easier to test the component (No need for fake dom in test setup).

```javascript
import { recordTracksEvent } from 'state/analytics/actions';

//...
this.props.recordTracksEvent( 'calypso_plans_autoconfig_click_manage_plans' );
//If we have a number of analytics calls, we can create a local variable:
const { recordTracksEvents: recordTracks } = this.props;
//...
//note that mapDispatchToProps can be a simple object if we don't need custom mapping
connect( mapStateToProps, { 
    recordTracksEvent 
} )( localize( PlansSetup ) );
```

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx [hunk 2]
```diff
@@ -263,73 +267,94 @@ const PlansSetup = React.createClass( {
...
+					</div>
+				);
+			case 'activate':
+			case 'configure':
+				return <Notice { ...statusProps } text={ translate( 'Almost done' ) } />;
+			case 'install':
+				return <Notice { ...statusProps } text={ translate( 'Working…' ) } />;
+			case 'wait':
+			default:
+				return <Notice { ...statusProps } text={ translate( 'Waiting to install' ) } />;
```
Comment (gwwar) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: I think the following is fine, but if we think that we'll always pass through the same status props, we can maybe move the switch into `getStatusText` where:

```javascript
renderStatus( plugin ) {
  if ( plugin.error ) {
    return this.renderStatusError( plugin );
  }

  const statusProps = {
  //...
  };

   return (
       <Notice { ...statusProps } text={ getStatusText( plugin ) } />
   );
}
```

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx [hunk 3]
```diff
@@ -263,73 +267,94 @@ const PlansSetup = React.createClass( {
...
-					break;
-				case 'wait':
-				default:
-					statusProps.text = translate( 'Waiting to install' );
-			}
+					) } />
+				);
+			default:
+				return (
+					<Notice { ...statusProps } text={ plugin.error.message || translate( 'An error occured.' ) } />
```
Comment (gwwar) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: Do we localize plugin.error.message? If not it might be better to leave a generic message, devs should still be able to see the error msg from the network tab.

**Comment by gwwar** [comment]
👍 Code changes look good here

**Comment by rickybanister** [comment]
I think this satisfies my concern, but I'd like to hear some other opinions before we merge.

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx (hunk 3 above)
Comment (gwwar) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: I'm not familiar with the endpoint, but unless the error messages are intended to be seen by a user, these can end up being rather cryptic for folks. It also avoids very funny cases where we might have a huge trace in the message. Anyway pretty minor, but I try to avoid these.

**Comment by beaulebens** [comment]
I agree with the previous comment about how we should ideally catch/avoid ever hitting this scenario, but in the case where a user finds themselves here, and hits this case, I think this handles it fine.

**Comment by gwwar** [comment]
Let's 🚢 then?
